### DISCUSSION TITLE
History of prisma migrations

### DISCUSSION THREAD

**Discussion by pedropmartiniano** [prisma/prisma#22660] [Q&A] [answered]
### Question

I wanna know how to generate a history of prisma migrations everytime when I run "npx prisma db pull" from an external database.

i'm using a mysql database, I created the database using the mysql workbench and all the changes that I will need to make will be made at the mysql workbench, and I need to generate a migration file like the history files of "npx prisma migrate dev", every time I run "npx prisma db pull", but I dont know how to do that, someone may help me with that?

**Comment by ludralph** [comment]
Hi @pedropmartiniano 👋 

Thank you for raising this question.

The `prisma db pull` command is used to introspect your database and generate or update your Prisma schema to reflect the current database schema. It does not create a migration history file like `npx prisma migrate dev` does. The migration history is typically generated when you create and apply migrations using Prisma Migrate, which tracks changes made through the Prisma schema.
If you are making changes directly in MySQL Workbench and want to reflect those changes in your Prisma migration history, you would typically need to run `prisma db pull` to update your Prisma schema and then create a migration file manually or by using `prisma migrate dev` to capture those changes in a new migration file. 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by pedropmartiniano** [reply]
So I can run `prisma db pull` and then run `prisma migrate dev` without change the schema to create the migration file and will be done what I want?

**Reply by ludralph** [reply, marked as answer]
Running `prisma db pull` followed by `prisma migrate dev` should ideally create a migration file based on the current database schema. The typical workflow for reflecting changes made directly in the database into Prisma migrations would involve the following steps:

1. Run `prisma db pull` to update your Prisma schema to match the current state of your database.
2. Manually adjust your Prisma schema if necessary to reflect any desired changes.
3. Run `prisma migrate dev` to create a new migration file based on the changes in your Prisma schema.

**Reply by pedropmartiniano** [reply]
Thank you!
